src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -179,31 +185,25 @@ public static int Log2(uint value)
             }
 
             // Already has contract 0->0, without branching
-            return Log2SoftwareFallback(value);
-        }
-
-        /// <summary>
-        /// Returns the integer (floor) log of the specified value, base 2.
-        /// Note that by convention, input value 0 returns 0 since Log(0) is undefined.
-        /// Does not incur branching.
-        /// </summary>
-        /// <param name="value">The value.</param>
-        private static int Log2SoftwareFallback(uint value)
-        {
-            // No AggressiveInlining due to large method size
-
-            value |= value >> 01;
-            value |= value >> 02;
-            value |= value >> 04;
-            value |= value >> 08;
-            value |= value >> 16;
+            return Log2Fallback(value);
 
-            // uint.MaxValue >> 27 is always in range [0 - 31] so we use Unsafe.AddByteOffset to avoid bounds check
-            return Unsafe.AddByteOffset(
-                // Using deBruijn sequence, k=2, n=5 (2^5=32) : 0b_0000_0111_1100_0100_1010_1100_1101_1101u
-                ref MemoryMarshal.GetReference(s_Log2DeBruijn),
-                // long -> IntPtr cast on 32-bit platforms is expensive - it does overflow checks not needed here
-                (IntPtr)(int)((value * 0x07C4ACDDu) >> 27));
+            int Log2Fallback(uint val)
```

Comment:
Local function, per @tannergooding suggestion in https://github.com/dotnet/coreclr/pull/22497

Reply:
nit: We've named it `SoftwareFallback` in other cases.

Reply:
Rolled back in latest commit, per follow-up advice on previous PR. 
We need to call it from 2 places